src/index.ml

```diff
@@ -651,6 +651,11 @@ struct
         Thread.return `Completed
     | Some witness -> merge ?hook ~witness t
 
+  let is_merging t =
+    let t = check_open t in
+    if t.config.readonly then raise RO_not_allowed;
+    Mutex.is_locked t.merge_lock
```

Comment:
Perhaps add a comment here to say that we know that `merge_lock` isn't held only by the asynchronous merge process, but we're considering it a good enough approximation for this case.